src/app/components/form/NumberField.tsx

```diff
@@ -0,0 +1,100 @@
+import { useRef } from "react";
+import { classNames } from "~/app/utils";
+
+export type Props = {
+  suffix?: string;
+  endAdornment?: React.ReactNode;
+  secondaryValue: string | number;
+  label: string;
+};
+
+export default function DualCurrencyInput({
+  label,
+  secondaryValue,
+  name,
+  id,
+  placeholder,
+  type = "text",
+  required = false,
+  pattern,
+  title,
+  onChange,
+  onFocus,
+  onBlur,
+  value,
+  autoFocus = false,
+  autoComplete = "off",
+  disabled,
+  min,
+  max,
+  suffix,
+  endAdornment,
+}: React.InputHTMLAttributes<HTMLInputElement> & Props) {
+  const inputEl = useRef<HTMLInputElement>(null);
+  const outerStyles =
+    "shadow-sm rounded-md border border-gray-300 dark:border-gray-800 bg-white dark:bg-black transition duration-300";
+
+  const inputNode = (
+    <input
+      ref={inputEl}
+      type={type}
+      name={id}
+      id={id}
+      className={classNames(
+        "block w-full placeholder-gray-500 dark:placeholder-gray-600 dark:text-white ",
+        "pr-0 border-0 focus:ring-0 bg-transparent"
+      )}
+      placeholder={placeholder}
+      required={required}
+      pattern={pattern}
+      title={title}
+      onChange={onChange}
+      onFocus={onFocus}
+      onBlur={onBlur}
+      value={value}
+      autoFocus={autoFocus}
+      autoComplete={autoComplete}
+      disabled={disabled}
+      min={min}
+      max={max}
+    />
+  );
+
+  return (
+    <div id="numberInput" className="relative block m-0">
```

Comment:
This shouldn't have an `id`. Also styles should not be applied to an `id`.  
If this field appears more than once in a document duplicated `id`s can lead to issues. For styling better use a class.